modules/web/src/app/backup/details/automatic-backup/component.ts

```diff
@@ -90,6 +93,13 @@ export class AutomaticBackupDetailsComponent implements OnInit, OnDestroy {
         this.backup = backup;
         this.isInitialized = true;
       });
+
+    this._projectService.selectedProject
+      .pipe(switchMap(project => this._clusterService.clusters(project.id, false)))
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(clusters => {
+        this.cluster = clusters.find(cluster => cluster.id === this.backup.spec.clusterId);
```

Comment:
you might end up here with a race condition.
You’re subscribing to clusters before you're sure `this.backup` is loaded.
It’d be safer to chain the cluster call with the backup load or combine them in one stream